docs/src/index.md

```diff
@@ -4,6 +4,350 @@ DocTestSetup = :(using ChainRulesCore, ChainRules)
+
+!!! note
+
+    The pullback returns one `∂arg` per `arg` to the original function, plus one for the fields of the function itself (explained below).
+
+!!! terminology
+    Sometimes _perturbation_, _seed_, and _sensitivity_ will be used interchangeably, depending on task/subfield (sensitivity analysis and perturbation theory are apparently very big on just calling everything _sensitivity_ or _perturbation_ respectively.)
+    At the end of the day, they are all _wiggles_ or _wobbles_.
+
+### Self derivative `Δself`, `∂self` etc.
+
+!!! terminology
+    To my knowledge there is no standard terminology for this.
+    Other good names might be `Δinternal`/`∂internal`.
+
+From the mathematical perspective, one may have been wondering what all this `Δself`, `∂self` is.
+After all a function with two inputs, say `f(a, b)`, only has two partial derivatives:
+``\dfrac{∂f}{∂a}``, ``\dfrac{∂f}{∂b}``.
+Why then does a `pushforward` take in this extra `Δself`, and why does a `pullback` return this extra `∂self`?
+
+The reason is that in Julia the function `f` may itself have internal fields.
+For example a closure has the fields it closes over; a callable object (i.e. a functor) like a `Flux.Dense` has the fields of that object.
+
+**Thus every function is treated as having the extra implicit argument `self`, which captures those fields.**
+So every `pushforward` takes in an extra argument, which is ignored unless the original function had fields.
+In is common to write `function foo_pushforward(_, Δargs...)` in the case when `foo` does not have fields.
+Similarly every `pullback` return an extra `∂self`, which for things without fields is the constant `NO_FIELDS`, indicating there are no fields within the function itself.
+
+#### Pushforward / Pullback summary
+
+- **Pushforward:**
+    - returned by `ffrule`
+    - takes input space wiggles, gives output space wobbles
+    - 1 argument per original function argument + 1 for the function itself
+    - 1 return per original function return
+
+- **Pullback**
+   - returned by `rrule`
+   - takes output space wobbles, gives input space wiggles
+   - 1 argument per original function return
+   - 1 return per original function argument + 1 for the function itself
+
+#### Pushforward/Pullback and Total Derivative/Gradient
+
+The most trivial use of `frule` and returned `pushforward` is to calculate the [Total Derivative](https://en.wikipedia.org/wiki/Total_derivative):
+
+```julia
+y, f_pushforward = frule(f, a, b, c)
+ẏ = f_pushforward(1, 1, 1, 1)  # for appropriate `1`-like perturbation.
+```
+
+Then we have that `ẏ` is the _total derivative_ of `f` at `(a, b, c)`, written mathematically as ``df_{(a,b,c)}``
+
+Similarly, the most trivial use of `rrule` and returned `pullback` is to calculate the [Gradient](https://en.wikipedia.org/wiki/Gradient):
+
+```julia
+y, f_pullback = rrule(f, a, b, c)
+∇f = f_pullback(1)  # for appropriate `1`-like seed.
+s̄, ā, b̄, c̄ = ∇f
```

Comment:
```suggestion
f̄, ā, b̄, c̄ = ∇f
```

Reply:
```suggestion
s̄elf, ā, b̄, c̄ = ∇f
```

Reply:
time to change `self` -> `φ` (`\varphi`)

Or you want to wait to change it in the code at the same time?